include/mlir/Dialect/AffineOps/AffineOps.h

```diff
@@ -522,6 +522,10 @@ bool isValidSymbol(Value *value);
 /// 2. drop unused dims and symbols from map
 void canonicalizeMapAndOperands(AffineMap *map,
                                 llvm::SmallVectorImpl<Value *> *operands);
+/// Canonicalizes an integer set the same way canonicalizeMapAndOperands does
+/// for affine maps.
+void canonicalizeSetAndOperands(IntegerSet *set,
+                                llvm::SmallVectorImpl<Value *> *operands);
```

Comment:
We should really be passing by reference here.

Reply:
canonicalizeMapAndOperands and canonicalizeSetAndOperands work in the same way, and if we change this one, it'd require changing the other one as well to be consistent and at the numerous places that's called from. I think it should only be done in another NFC PR. 

On another note, the ampersands really help with readability since they indicate output parameters, and make it obvious at the call site that the passed things are being changed. If we'd like to move to pass by reference for all non-nullable inputs or outputs consistently, I think this should be stated in the style guide. 

```
auto map = ...
canonicalizeMapAndOperands(&map, &operands);
...
```

Reply:
Done in 57ebdf105583cb39bd8d29b83dc50dd0cf50868e